packages/react/src/lib/accessibility/Behaviors/Dialog/dialogBehavior.ts

```diff
@@ -15,12 +16,50 @@ import popupFocusTrapBehavior from '../Popup/popupFocusTrapBehavior'
  */
 const dialogBehavior: Accessibility = (props: any) => {
   const behaviorData = popupFocusTrapBehavior(props)
+  const defaultAriaLabelledBy = getDefaultAriaLabelledBy(props)
+  const defaultAriaDescribedBy = getDefaultAriaDescribedBy(props)
   behaviorData.attributes.popup = {
     ...behaviorData.attributes.popup,
     role: 'dialog',
+    'aria-labelledby': defaultAriaLabelledBy || props['aria-labelledby'],
+    'aria-describedby': defaultAriaDescribedBy || props['aria-describedby'],
+  }
+  behaviorData.attributes.header = {
+    id: defaultAriaLabelledBy,
+  }
+  behaviorData.attributes.content = {
+    id: defaultAriaDescribedBy,
   }
 
   return behaviorData
 }
 
+/**
+ * Returns the element id of the header or generates a default one. Is
+ * used when user does not provide aria-label or aria-labelledby as
+ * props. It is also used as default value for header id if there is not
+ * any value provided by user as prop.
+ */
+const getDefaultAriaLabelledBy = (props: any) => {
+  const { header } = props
+  if (props['aria-label'] || props['aria-labelledby'] || !header) {
+    return undefined
+  }
+  return header['id'] || _.uniqueId('dialog-header-')
+}
+
+/**
+ * Returns the element id of the header or generates a default one. Is
```

Comment:
`id of the content`

Reply:
that's what happens when you copy paste without thinking, kids.